### ISSUE TITLE
Unsecure default access rights for users.xml

### ISSUE THREAD

**Issue by sokolov-denis** [ClickHouse/ClickHouse#588]
**OS:** Ubuntu 16.04
**Clickhouse:** 1.1.54181

After installation from packages any user can read users.xml:
```
$ stat /etc/clickhouse-server/users.xml
  File: '/etc/clickhouse-server/users.xml'
  Size: 3857            Blocks: 8          IO Block: 4096   regular file
Device: 801h/2049d      Inode: 258453      Links: 1
Access: (0644/-rw-r--r--)  Uid: (  112/clickhouse)   Gid: (  117/clickhouse)
Access: 2017-03-14 08:26:52.573261233 +0000
Modify: 2017-03-09 18:15:45.000000000 +0000
Change: 2017-03-14 08:26:18.672319215 +0000
 Birth: -

$ grep password /etc/clickhouse-server/users.xml
```

**Comment by proller** [comment]
second problem: 
-rw-rw-rw-   1 clickhouse clickhouse   4906 Feb 13 00:32 users-preprocessed.xml

**Comment by sokolov-denis** [comment]
> second problem:
> -rw-rw-rw- 1 clickhouse clickhouse 4906 Feb 13 00:32 users-preprocessed.xml

CH creates **all** files/directories with access rights 666/777

https://github.com/yandex/ClickHouse/blob/v1.1.54181-stable/contrib/libpoco/Foundation/src/File_UNIX.cpp#L385
https://github.com/yandex/ClickHouse/blob/v1.1.54181-stable/contrib/libpoco/Foundation/src/File_UNIX.cpp#L405

**Comment by alexey-milovidov** [comment]
This was fixed long time ago.

```
$ ls -l /etc/clickhouse-server/users-preprocessed.xml 
-rw-r--r-- 1 clickhouse clickhouse 5122 окт.  17 07:07 /etc/clickhouse-server/users-preprocessed.xml
```

**Comment by alexey-milovidov** [comment]
@proller This is NOT fixed.
```
root@ca26ee8df9c2:/# ls -l /etc/clickhouse-server/
total 28
-rw-r--r-- 1 root root 16993 Jan 23 22:21 config.xml
lrwxrwxrwx 1 root root    40 Jan 26 23:31 preprocessed -> /var/lib/clickhouse/preprocessed_configs
-rw-r--r-- 1 root root  4744 Jan 23 22:21 users.xml
```

**Comment by alexey-milovidov** [comment]
Fixed in #22976:

```
milovidov@github-explorer:~$ sudo ls -l /etc/clickhouse-server/
total 68
dr-x------ 2 clickhouse clickhouse  4096 Dec  1  2020 config.d
-r-------- 1 clickhouse clickhouse 38445 Dec  7  2020 config.xml
-rw-r--r-- 1 clickhouse clickhouse  1679 Dec  3  2020 server.crt
-rw------- 1 clickhouse clickhouse  1708 Dec  3  2020 server.key
dr-x------ 2 clickhouse clickhouse  4096 Dec  1  2020 users.d
-r-------- 1 clickhouse clickhouse 11390 Mar 18 19:31 users.xml
```
